### DISCUSSION TITLE
LS-QL upgrade choices

### DISCUSSION THREAD

**Discussion by markblinkhorn** [1000001101000/Debian_on_Buffalo#220] [Q&A] [answered]
I've been running buster on a LS-QL (armel) device for a couple of years now. It was installed using your excellent guidance - for which, thank you.
The device sits on my private network and is not accessible from the wider internet and never will be. I recently had an issue with 'frragment errors' spamming the syslog which prompted me to try running an apt update. At this point I discovered that armel support has been withdrawn from the buster repo.

I note that you generally discourage dist-upgrade, so I guess I have only a couple of choices:
    1. put up with it
    2. fresh install a later version

My question is - my data is contained in an LVM Virtual Group on /dev/sdx3, I can (and will) back it up before I begin, but what's the chance of it surviving the process? Restoring all of that data will take a couple of aeons (probably geological in magnitude) on this device.

;)

**Comment by 1000001101000** [comment, marked as answer]
Good to hear from you!

Some thoughts in no particular order:

- As Buster's support status with debian ends it's normal for them to move the repo to the "archive". If you update your apt config to use https://archive.debian.org/debian/ it should resume working, though probably without further updates. 
- At this point in Buster's lifecycle I wouldn't expect an apt-update to resolve any new errors in the log, you should probably investigate that first since something like a disk/volume/filesystem issue would only get worse trying to do a dist-upgrade.
- Buster was pretty long ago at this point, my memory of the state of my customization of the various components from then is far from perfect. Unfortunately I haven't done the work of making those pieces modular so they can be upgraded. 
- The LS-QL has the most potential failure points for a dist-upgrade of any of the devices supported in the project. 
- I've not used LVM much and never on these devices so my experience is limited. 
- You should take a look at SMART data etc and ensure your disks are healthy before proceeding with anything.

My best suggestion for upgrading:

- Copy the installer files to /boot similar to how you would if you were starting from the stock firmware and run through a normal install and save setting up the LVM pieces once you boot into the newer Debian version. 
- If that ends up failing and you're faced with doing a clean install and reloading your data from backup try using the deboostrap method to create a bootable disk image rather than waiting through the installer.

**Reply by markblinkhorn** [reply]
Thanks for those thoughts - they are helpful. I will carefully consider my next steps before throwing caution to the winds! I'll let you know. ;)

**Reply by markblinkhorn** [reply]
Failed at first hurdle:
E: The repository 'https://archive.debian.org/debian-security buster/updates Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.

And it is true:- https://archive.debian.org/debian-security/dists/ does not conain a buster folder...

**Comment by 1000001101000** [comment]
Yep. no more updates, security, backports etc once it reaches that point. Truly an archive.
